libraries/c_sdk/aws/defender/src/aws_iot_defender_collector.c

```diff
@@ -33,20 +33,35 @@
 
 #include "platform/iot_clock.h"
 
+#include "aws_iot_defender_schema_v1.h"
+
+/* FreeRTOS specific. -- todo: this file should not take dependency on FreeRTOS.  */
+#include "task.h"
+
 /* Used in debugging. It will decode the report with cbor format and print to stdout. */
-#define DEBUG_CBOR_PRINT    0
+#define DEBUG_CBOR_PRINT                0
+
+
+/* Magic numbers. */
+#if ( configDefenderReportHeapStats == 1 )
+    #define MAGIC_NUMBER_HEAP_STATS    ( 7 )      /* If heap metrics group is enabled by user. */
+#else
+    #define MAGIC_NUMBER_HEAP_STATS    ( 0 )
+#endif /* configDefenderReportHeapStats */
 
-#define  HEADER_TAG         AwsIotDefenderInternal_SelectTag( "header", "hed" )
-#define  REPORTID_TAG       AwsIotDefenderInternal_SelectTag( "report_id", "rid" )
-#define  VERSION_TAG        AwsIotDefenderInternal_SelectTag( "version", "v" )
-#define  VERSION_1_0        "1.0"   /* Used by defender service to indicate the schema change of report, e.g. adding new field. */
-#define  METRICS_TAG        AwsIotDefenderInternal_SelectTag( "metrics", "met" )
+#if ( configDefenderReportRuntimeStats == 1 )
```

Comment:
Hmm, the use case for the macros is user suppose to configure what metrics to use in compile time in ```FreeRTOSConfig.h```. Since there's hard dependency on port configuration for kernel metrics that the configuration is not demo configuration anymore. (To be specific, which heap_*.c is used has impact on heap metrics. Task configuration may have impact on runtime stats. So that the limits are decided in compile time by port configuration.)

Maybe I'm not answering your questions directly, which naming convention are you suggesting?